packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationMessagingService.java

```diff
@@ -167,4 +168,54 @@ private Bundle convertDataToBundle(RemoteMessage message) {
         return bundle;
     }
 
+	/**
+	 * Called if InstanceID token is updated. This may occur if the security of
+	 * the previous token had been compromised. Note that this is called when the InstanceID token
+	 * is initially generated so this is where you would retrieve the token.
+	 */
+	@Override
+	public void onNewToken(String refreshedToken) {
+		// Get updated InstanceID token.
+		Log.i(LOG_TAG, "Refreshed token: " + refreshedToken);
+
+		final Bundle bundle = createBundleWithToken(refreshedToken);
+		// We need to run this on the main thread, as the React code assumes that is true.
+		// Namely, DevServerHelper constructs a Handler() without a Looper, which triggers:
+		// "Can't create handler inside thread that has not called Looper.prepare()"
+		Handler handler = new Handler(Looper.getMainLooper());
+		handler.post(new Runnable() {
+			public void run() {
+				// Construct and load our normal React JS code bundle
```

Comment:
Do we need JS there? 

```suggestion
				// Construct and load our normal React code bundle
```